include/exadg/operators/operator_base.h

```diff
@@ -320,6 +318,19 @@ class OperatorBase : public dealii::Subscriptor
   void
   initialize_block_diagonal_preconditioner_matrix_free() const;
 
+  // updates block diagonal preconditioner when using the matrix-free variant with elementwise
+  // iterative solvers and matrix-free operator evaluation.
+  void
+  update_block_diagonal_preconditioner_matrix_free() const;
+
+  // initializes block diagonal preconditioner for matrix-based variant
```

Comment:
lines 326 & 328: Can we add something like "using cell-wise solves." to make it a bite more verbose?

Reply:
this is actually the matrix-based variant, where "solves" might be irritating as well, or at least not precise enough to differentiate from "cell-wise iterative solves"?

Reply:
Ah, so the matrix-based variant always does `calculate_lu_factorization_block_jacobi(matrices);`, while the matrix-free application of the inverse mass matrix *and* the element Krylov solvers both follow the matrix-free path?
... then you are right ... maybe add "using cell-wise LU decomposition"?